### PR TITLE
Allow using it on fzf preview window

### PR DESCRIPTION
**Pull request by dan-santana**
In order to make it work in the fzf preview window, I had to make some changes:
* Add an exit code on errors, so I can fallbacks when `timg` fails e.g. `timg $1 2> /dev/null || file -i $1`
* Disable cursor hiding, as it seems fzf doesn't like some escape sequences and will print them as ASCII text in the preview window
* Limit the amount of frames rendered on animations, so it can show only the first frame (fzf doesn't like streams on preview too)

And the final result is:
screenshot-2018-04-22t01 25 33-0300

It's running on rxvt-unicode (with a few patches, including 24-bit colors) and using Base16 Monokai colorscheme (that's the reason for the black borders on the transparent parts of the image).

I also added an example on how to use it to preview images with fzf on README.md, but didn't know where to put the image above... so feel free to put it if you want to :)

### CODE REVIEW

File: src/timg.cc [hunk 1]
```diff
@@ -160,10 +160,15 @@ static bool LoadImageAndScale(const char *filename,
 }
 
 void DisplayAnimation(const std::vector<Magick::Image> &image_sequence,
-                      tmillis_t duration_ms, int loops, int w, int h, int fd) {
+                      tmillis_t duration_ms, int max_frames, int loops,
+                      int w, int h, int fd) {
     std::vector<PreprocessedFrame*> frames;
+    if (max_frames == -1) {
+        max_frames = image_sequence.size();
```
[the remark this answers is not in the document]
Comment (dan-santana) [reply src/timg.cc]: It would segfault, indeed.

[the remark this answers is not in the document]
Comment (dan-santana) [reply src/timg.cc]: Fixed!

**Comment by hzeller** [comment]
If the user give `-f-2`, they will not see anything, but I guess that is essentially expected.

**Comment by hzeller** [comment]
Anyway, merged. Thanks for your contribution!

**Pull request merged**